src/main/java/com/iexec/core/replicate/ReplicatesService.java

```diff
@@ -216,153 +216,187 @@ public void updateReplicateStatus(String chainTaskId,
-                log.error(getReplicateFieldsErrorLog("resultLink missing", chainTaskId, walletAddress));
-                return null;
-            }
+    private boolean canUpdateToUploadSuccess(String chainTaskId, Replicate replicate,
+                                             ReplicateStatusUpdate statusUpdate) {
+        ReplicateStatus newStatus = statusUpdate.getStatus();
+        ReplicateStatusDetails details = statusUpdate.getDetails();
 
-            replicate.setResultLink(details.getResultLink());
-            replicate.setChainCallbackData(details.getChainCallbackData());
+        if (!verifyStatus(chainTaskId, replicate.getWalletAddress(), newStatus)) {
+            log.error("Cannot update replicate, status is not correct",
+                    getLogDetails(chainTaskId, replicate, statusUpdate));
+            return false;
         }
 
-        replicate.updateStatus(statusUpdate);
-        return replicate;
-    }
+        if (details == null || details.getResultLink() == null || details.getResultLink().isEmpty()) {
+            log.error("Cannot update replicate, missing resultLink {}",
+                    getLogDetails(chainTaskId, replicate, statusUpdate));
+            return false;
+        }
 
-    private String getReplicateStatusErrorLog(String error, String chainTaskId, String walletAddress, ReplicateStatus currentStatus, ReplicateStatus newStatus, ReplicateStatusModifier modifier) {
-        return getUpdateReplicateStatusLogMessage("Failed to updateReplicateStatus", error, chainTaskId, walletAddress, currentStatus, newStatus, modifier);
-    }
+        replicate.setResultLink(details.getResultLink());
+        replicate.setChainCallbackData(details.getChainCallbackData());
+        return true;
 
-    private String getReplicateStatusWarningLog(String warning, String chainTaskId, String walletAddress,
-                                                ReplicateStatus currentStatus, ReplicateStatus newStatus,
-                                                ReplicateStatusModifier modifier) {
-        return getUpdateReplicateStatusLogMessage("Replicate status updated", warning, chainTaskId, walletAddress, currentStatus, newStatus, modifier);
     }
 
-    private String getReplicateStatusSuccessLog(String chainTaskId, String walletAddress, ReplicateStatus currentStatus, ReplicateStatus newStatus, ReplicateStatusModifier modifier) {
-        return getUpdateReplicateStatusLogMessage("UpdateReplicateStatus succeeded", "", chainTaskId, walletAddress, currentStatus, newStatus, modifier);
+    private boolean verifyStatus(String chainTaskId, String walletAddress, ReplicateStatus status) {
+        switch (status) {
+            case CONTRIBUTED:
+                return iexecHubService.isStatusTrueOnChain(chainTaskId, walletAddress,
+                        ChainContributionStatus.CONTRIBUTED);
+            case CONTRIBUTE_FAILED:
```

Comment:
I think we don't want to update to CONTRIBUTE_FAILED & REVEAL_FAILED anymore

Reply:
Why not? the worker may not succeed to contribute or reveal (tx issues or not correct hash)!

Reply:
As a worker, are you able to provide me a proof of a failed contribution?
If yes, so we can keep this status

Reply:
Well it's the same also for the result upload!